### ISSUE TITLE
Feature request: let the pattern If ... fi handle consequence/alternative of type field[n]

### ISSUE THREAD

**Issue by GuthL** [Zokrates/ZoKrates#182]
Having the if pattern handle such elements would be nice
```
def main() -> (field):
	field a = 0
	field[2] arr = [1,1]
	field[2] other = [2,2]

	field[2] result = [0,0]

    result = if a == 0 then arr else other fi

    result == arr

    return 1

```
The current implementation returns a semantic error.  On a side note, I liked the error message for being self-explanatory.
```
thread 'main' panicked at 'Compilation failed: Semantic error: if {condition} then {consequence} else {alternative} should have types bool, field, field, found bool, field[2], field[2]', zokrates_cli/src/bin.rs:217:29
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by Schaeff** [comment]
Agreed! This would be good to have, and should not be too hard to implement based on the `field` case.

**Comment by Schaeff** [comment]
This got addressed in #225
